### PR TITLE
reconciler: improve variable names and extract methods from inline logic

### PR DESCRIPTION
**Pull request by DerekStrickland**
This PR includes readability improvements to `scheduler.allocReconciler` and `scheduler.allocSet`. It does not change behavior.

### CODE REVIEW

File: scheduler/reconcile.go [hunk 1]
```diff
@@ -190,8 +189,8 @@ func (a *allocReconciler) Compute() *reconcileResults {
 	// Create the allocation matrix
 	m := newAllocMatrix(a.job, a.existingAllocs)
 
-	// Handle stopping unneeded deployments
-	a.cancelDeployments()
+	// Stop unneeded deployments
+	a.cancelUnneededDeployments()
```
Comment (tgross) [inline scheduler/reconcile.go:193]: If changing the name didn't obviate the comment, then it's the wrong name 😁 

```suggestion
	a.cancelUnneededDeployments()
```

File: scheduler/reconcile.go [hunk 2]
```diff
@@ -200,30 +199,28 @@ func (a *allocReconciler) Compute() *reconcileResults {
...
-		// state. But we also need to tell Compute we're paused, otherwise we
-		// make placements on the paused deployment.
-		if a.job.IsMultiregion() && !(a.job.IsPeriodic() || a.job.IsParameterized()) {
-			a.deploymentPaused = true
-		}
-	}
+	a.computeDeploymentPaused()
 
 	// Reconcile each group
+	deploymentComplete := a.computeDeploymentComplete(m)
```
Comment (tgross) [inline scheduler/reconcile.go:205]: Now that we've moved `computeGroup` behind the function call, this comment isn't as clear. 

```suggestion
	// Reconcile each group to determine if the deployment is complete
	deploymentComplete := a.computeDeploymentComplete(m)
```

Alternately (maybe preferably?), just cut the comment and leave the method name to do the work of communicating what's happening.

**Review by tgross — APPROVED** [review]
LGTM with a couple comment cleanups.

File: scheduler/reconcile.go [hunk 3]
```diff
@@ -200,30 +199,28 @@ func (a *allocReconciler) Compute() *reconcileResults {
...
-		// When we create the deployment later, it will be in a pending
-		// state. But we also need to tell Compute we're paused, otherwise we
-		// make placements on the paused deployment.
-		if a.job.IsMultiregion() && !(a.job.IsPeriodic() || a.job.IsParameterized()) {
-			a.deploymentPaused = true
-		}
-	}
+	a.computeDeploymentPaused()
 
 	// Reconcile each group
```
Comment (DerekStrickland) [inline scheduler/reconcile.go:203]:
```suggestion
```

**Pull request merged**